src/react/utils/hooks.ts

```diff
@@ -186,6 +187,8 @@ export const useAccounts = () => {
           Name: current.Name,
           CreationDate: current.CreationDate,
           Roles: [...(agg[current.Name]?.Roles || []), ...current.Roles],
+          // retrieve the account id from the role ARN
+          id: current.Roles[0].Arn.split(':')[4],
```

Comment:
nit: I would suggest parsing the ARN with a regex to get clearer and more resiliently the Id.

`arn:aws:iam::000000000000:role/scality-internal/storage-manager-role` could be parsed using

`/arn:aws:iam::(?<account_id>/d+):/(?<role_name>.+)/` then you can retrieve the id using the `account_id` capture group.

Reply:
Yes...for sure we should rely on regex!
Fixed here: 
https://github.com/scality/zenko-ui/pull/334/commits/0129c206bfe2b2db6840033ad8030efeebf2fb5f#diff-34f23db3cb2d457a06d84e4648f2ffe766e7f9b16d44c389f4e766bf2fa514a1R147